rdt-open-reader/src/main/java/io/ona/rdt/util/Utils.java

```diff
@@ -225,4 +228,42 @@ public static boolean isValidJSONObject(String str) {
             return false;
         }
     }
+
+    public static void verifyUserAuthorization() {
+        UserAuthorizationVerificationTask.getInstance().run();
```

Comment:
Here, check whether the asynctask is pending or finished. If pending, execute it, if finished, destroy it, get a new instance and execute it.

Reply:
done